### PR TITLE
Adding type definitions for multi-progress

### PR DESCRIPTION
**Pull request by DHBrett**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/multi-progress/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+// Project: https://github.com/pitaj/multi-progress
+// Definitions by: David Brett <https://github.com/DHBrett>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node"/>
+
+import ProgressBar, { ProgressBarOptions } from '../progress'
+import { Stream } from 'stream';
+
+declare class MultiProgress {
```
Comment (amcasey) [inline types/multi-progress/index.d.ts]: Did you mean to export `MultiProgress`?

File: types/multi-progress/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{
+    "compilerOptions": {
+        "module": "commonjs",
+        "lib": [
+            "es6"
+        ],
+        "noImplicitAny": true,
+        "noImplicitThis": true,
+        "strictNullChecks": false,
```
Comment (amcasey) [inline types/multi-progress/tsconfig.json]: Why are strictNullChecks disabled?

File: types/multi-progress/tslint.json [hunk 3]
```diff
@@ -0,0 +1,79 @@
+{
+    "extends": "dtslint/dt.json",
+    "rules": {
+        "adjacent-overload-signatures": false,
```
Comment (amcasey) [inline types/multi-progress/tslint.json]: Why are the lint rules disabled?

**Review by amcasey — CHANGES_REQUESTED** [review]

File: types/multi-progress/index.d.ts (hunk 1 above)
Comment (DHBrett) [reply types/multi-progress/index.d.ts]: Yep, my mistake

File: types/multi-progress/tsconfig.json (hunk 2 above)
Comment (DHBrett) [reply types/multi-progress/tsconfig.json]: I just copied these in from another folder in the project (progress), I've turned it back on.

File: types/multi-progress/tslint.json (hunk 3 above)
Comment (DHBrett) [reply types/multi-progress/tslint.json]: Again, my mistake for copying these without checking whether they were actually needed. Fixed.

**Review by amcasey — APPROVED** [review]

File: types/multi-progress/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
+// Type definitions for multi-progress 2.0
+// Project: https://github.com/pitaj/multi-progress
+// Definitions by: David Brett <https://github.com/DHBrett>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node"/>
+
+import ProgressBar, { ProgressBarOptions } from '../progress';
```
Comment (sandersn) [inline types/multi-progress/index.d.ts]: this is incorrect. should be just 'progress'

File: types/progress/index.d.ts [hunk 5]
```diff
@@ -138,4 +138,4 @@ declare class ProgressBar {
 }
 declare namespace ProgressBar { }
 
-export = ProgressBar;
+export default ProgressBar;
```
Comment (bkendall) [inline types/progress/index.d.ts]: Is this change correct? I have a `ts` file with the following that used to work that is now broken (and reverting `@types/progress` is the fix)

```typescript
import * as ProgressBar from "progress";
new ProgressBar(":bar");
```

`progress` doesn't have a `default` export, so this is incorrect

Comment (sandersn) [reply types/progress/index.d.ts]: Fix is up at #31398

**Pull request merged**